### PR TITLE
[REF] Cleanup billingBlock.js

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Code cleanup in `billingBlock.js`

Before
----------------------------------------
Code messier - unnecessary functions, deprecated variables, malformed html

After
----------------------------------------
Code neater

### CODE REVIEW

File: templates/CRM/Core/BillingBlock.js [hunk 1]
```diff
@@ -1,71 +1,62 @@
...
-  /**
-   * Adds the icons of enabled credit cards
-   * Handles clicking on a icon.
-   * Changes the icon depending on the credit card number.
-   * Removes spaces and dashes from credit card numbers.
-   */
-  function civicrm_billingblock_creditcard_helper() {
-    $(function() {
-      $.each(CRM.config.creditCardTypes, function(key, val) {
-        var html = '<a href="#" data-card_type=" + key + " title="' + val + '" class="crm-credit_card_type-icon-' + val.css_key + '"><span>' + val.label + '</span></a>';
```
Comment (colemanw) [inline templates/CRM/Core/BillingBlock.js:13]: Note `data-card_type` was never set correctly due to bad js concatenation here, so it's obviously unused and not needed.

File: templates/CRM/Core/BillingBlock.js [hunk 2]
```diff
@@ -1,71 +1,62 @@
...
+      $.each(card_values, function(key, spec) {
+        if (ccnumber.match('^' + spec.pattern + '$')) {
+          $('.crm-container .credit_card_type-section .crm-credit_card_type-icon-' + spec.css_key).css('opacity', 1);
+          $('select#credit_card_type').val(key);
+          return false;
+        }
+      });
+    }
 
-  civicrm_billingblock_creditcard_helper();
```
Comment (colemanw) [inline templates/CRM/Core/BillingBlock.js:69]: This was silly. The function just called another function.

**Comment by braders** [comment]
This looks much better 👍

**Comment by seamuslee001** [comment]
This has been merge ready for a few days merging now

**Pull request merged**